dlib/media/ffmpeg_muxer.h

```diff
@@ -916,13 +916,23 @@ namespace dlib
                     args.args_image.h = num_rows(image);
                     args.args_image.w = num_columns(image);
                     args.args_image.framerate = 1;
-                    args.format_options["update"] = "1";
-                    if (pixel_traits<typename image_type::value_type>::has_alpha)
-                        args.args_image.codec_options["fmt"] = "rgba";
+                    args.args_image.codec_options = codec_options;
+                    using traits = pixel_traits<typename image_type::value_type>;
+                    if (traits::bgr_layout)
+                    {
+                        if (traits::has_alpha)
+                            args.args_image.fmt = AV_PIX_FMT_ABGR;
+                        else
+                            args.args_image.fmt = AV_PIX_FMT_BGR24;
+                    }
```

Comment:
This isn't required.

Reply:
If I don't specify the pixel format or the framerate, ffmpeg refuses to process the image.

Reply:
By default, ```encoder_image_args::fmt``` is set to ```AV_PIX_FMT_YUV420P```. It will use that by default. If the codec doesn't support that it will use something that it does. You can try set it to the same format as the input image, as you're doing, but there is no guarantee that the codec supports that and that it will be used. I guess by default, you should try use the same format as your input image.

Reply:
> If I don't specify the pixel format or the framerate, ffmpeg refuses to process the image.

Yeah you're right.